pkg/registry/discovery/endpointslice/storage/storage.go

```diff
@@ -32,16 +35,15 @@ type REST struct {
 // NewREST returns a RESTStorage object that will work against endpoint slices.
 func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, error) {
 	store := &genericregistry.Store{
-		NewFunc:     func() runtime.Object { return &discovery.EndpointSlice{} },
-		NewListFunc: func() runtime.Object { return &discovery.EndpointSliceList{} },
-		ObjectNameFunc: func(obj runtime.Object) (string, error) {
-			return obj.(*discovery.EndpointSlice).Name, nil
-		},
+		NewFunc:                  func() runtime.Object { return &discovery.EndpointSlice{} },
+		NewListFunc:              func() runtime.Object { return &discovery.EndpointSliceList{} },
 		DefaultQualifiedResource: discovery.Resource("endpointslices"),
 
 		CreateStrategy: endpointslice.Strategy,
 		UpdateStrategy: endpointslice.Strategy,
 		DeleteStrategy: endpointslice.Strategy,
+
+		TableConvertor: printerstorage.TableConvertor{TableGenerator: printers.NewTableGenerator().With(printersinternal.AddHandlers)},
```

Comment:
@soltysh, when you get a chance, is this the right construct?

Reply:
All of the other resources have the same `TableConvertor` construct (using the `TableGenerator`). `AddHandlers` adds the printing function for `EndpointSlice`:

https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L1169